### DISCUSSION TITLE
Pulling kafka/service registry validators into SDK

### DISCUSSION THREAD

**Discussion by wtrocki** [redhat-developer/app-services-cli#731] [Ideas]
Hi

I have noticed that we have some of the validation functions nicely organized in the kafka/registry folders.
It will be good if we actually make them part of the SDK for others to reuse - like kafka name/registry name etc.

Same could be good for status code/error code mappings etc.

**Comment by wtrocki** [comment]
Related: 
https://github.com/bf2fc6cc711aee1a0c2a/srs-fleet-manager/issues/61
https://github.com/pmuir/terraform-provider-rhoas

CC @craicoverflow @pmuir

**Comment by craicoverflow** [comment]
> I have noticed that we have some of the validation functions nicely organized in the kafka/registry folders.

These validators in their current form are highly specific to Cobra APIs. 

We could pull them into the SDKs but simplified a bit (there would be no need to pass an `interface{}` type - then the Cobra validators could wrap these.

**Comment by craicoverflow** [comment]
> Same could be good for status code/error code mappings etc.

I think that the error helpers are specific to the CLI and probably the currently returned error is enough for the client to handle it in its own way.
